homeassistant/components/system_bridge/media_player.py

```diff
@@ -0,0 +1,273 @@
+    """Define a System Bridge media player."""
+
+    entity_description: MediaPlayerEntityDescription
+
+    def __init__(
+        self,
+        coordinator: SystemBridgeDataUpdateCoordinator,
+        description: MediaPlayerEntityDescription,
+        api_port: int,
+    ) -> None:
+        """Initialize."""
+        super().__init__(
+            coordinator,
+            api_port,
+            description.key,
+        )
+        self.entity_description = description
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.coordinator.data.media is not None
+
+    @property
+    def supported_features(self) -> MediaPlayerEntityFeature:
+        """Flag media player features that are supported."""
+        features = (
+            MediaPlayerEntityFeature.REPEAT_SET | MediaPlayerEntityFeature.SHUFFLE_SET
+        )
+
+        data: SystemBridgeCoordinatorData = self.coordinator.data
+        if data.media.is_previous_enabled:
+            features |= MediaPlayerEntityFeature.PREVIOUS_TRACK
+        if data.media.is_next_enabled:
+            features |= MediaPlayerEntityFeature.NEXT_TRACK
+        if data.media.is_pause_enabled:
+            features |= MediaPlayerEntityFeature.PAUSE
+        if data.media.is_play_enabled:
+            features |= MediaPlayerEntityFeature.PLAY
+        if data.media.is_stop_enabled:
+            features |= MediaPlayerEntityFeature.STOP
+
+        return features
+
+    @property
+    def state(self) -> MediaPlayerState | None:
+        """State of the player."""
+        data: SystemBridgeCoordinatorData = self.coordinator.data
+        if data.media.status is None:
+            return None
+        return MEDIA_STATUS_MAP.get(
+            data.media.status,
+            MediaPlayerState.IDLE,
+        )
+
+    @property
+    def media_duration(self) -> int | None:
+        """Duration of current playing media in seconds."""
+        data: SystemBridgeCoordinatorData = self.coordinator.data
+        if data.media.duration is None:
+            return None
+        return int(data.media.duration)
+
+    @property
+    def media_position(self) -> int | None:
+        """Position of current playing media in seconds."""
+        data: SystemBridgeCoordinatorData = self.coordinator.data
+        if data.media.position is None:
+            return None
+        return int(data.media.position)
+
+    @property
+    def media_position_updated_at(self) -> dt.datetime | None:
+        """When was the position of the current playing media valid."""
+        data: SystemBridgeCoordinatorData = self.coordinator.data
+        if data.media.updated_at is None:
+            return None
+        return dt.datetime.fromtimestamp(data.media.updated_at)
+
+    @property
+    def media_title(self) -> str | None:
+        """Title of current playing media."""
+        data: SystemBridgeCoordinatorData = self.coordinator.data
```

Comment:
You can create a separate property for self.coordinator.data, saves adding this line every method